### PR TITLE
Create user-authorization.md

### LINKED ISSUE #10521: Document User authorization patterns
There are multiple ways we authorize users.

* CSUM functions https://github.com/department-of-veterans-affairs/appeals-deployment/blob/master/ansible/vars/functions-config.yml
* User roles

Create a `docs/user-authorization.md` document describing how/why both patterns exist and the proper one to choose for a particular use case.

### PR DESCRIPTION
**Pull request by laurjpeterson**
Resolves #10521

### Description
Documents current state of user authorization

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Go to ...

### Documentation Updates
- [ ] Topline File Descriptions Added or Updated as Needed

*Schema Changes Only*

* [ ] Column comments updated
* [ ] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [ ] Appropriate indexes added (especially for foreign keys, polymorphic columns, and unique constraints)
* [ ] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: docs/user-authorization.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+TODO: comb through each and
+- identify which can be removed because all users can have the functionality
+- identify which should be replaced with longer term permissioning
+  - LP's opinion:
+    - `withdraw_decision_review`
+    - `remove_decision_reviews`
+
+## VACOLS
+- The VACOLS staff table is the canonical board list of folks in many roles (Attorney, Judge, Co-located), and it currently is role-based
```
Comment (alisan16) [inline docs/user-authorization.md]: What are these VACOLS permissions used for in Caseflow? Would be good to add a sentence for context similar to the feature examples in the sections for the other permission types.

**Comment by pkarman** [comment]
this document is really good. I'm not sure it belongs in the caseflow repo yet. There are too many unanswered questions. Things in the repo are public and used as reference docs. This feels like maybe it should be a wiki page at first, till we sort out the unanswered/unclear parts, and then transfer to the repo when we feel it's a solid description of the authorization model.

**Comment by laurjpeterson** [comment]
@pkarman I completely agree that this shouldn't be in the caseflow repo. Just did it here because of the PR and commenting features.

How about we let it go through a few more cycles of feedback, then we can transfer it to the wiki. I agree that the wiki should be fore technical documentation.

**Comment by lowellrex** [comment]
Looks good to my eyes! Thanks for writing this all down @laurjpeterson !

**Comment by laurjpeterson** [comment]
@lowellrex - could you fill in the blanks around how we rely on VACOLS roles for judge, attorney, and AVLJ?

**Comment by pkarman** [comment]
coped to wiki https://github.com/department-of-veterans-affairs/caseflow/wiki/User-authorizations-in-Caseflow

**Pull request closed without merging**